modules/sor/sorV2/lib/pools/weighted/weightedPool.ts

```diff
@@ -175,6 +188,83 @@ export class WeightedPool implements BasePool {
         return tokenInAmount;
     }
 
+    public addLiquiditySingleTokenExactIn(tokenIn: Token, bpt: Token, amount: TokenAmount): TokenAmount {
+        try {
+            // balances and amounts must be normalized to 1e18 fixed point - e.g. 1USDC => 1e18 not 1e6
+            const { tokenBalances, amountsIn, weights } = Array.from(this.tokenMap.values()).reduce(
+                (acc: { tokenBalances: bigint[]; amountsIn: bigint[]; weights: bigint[] }, weightedPoolToken) => {
+                    if (weightedPoolToken.token.address.toLowerCase() === tokenIn.address.toLowerCase()) {
```

Comment:
You should be able to use `isSameAddress`:
```suggestion
                    if (weightedPoolToken.token.isSameAddress(tokenIn.address) {
```